src/plugins/intel_gpu/include/intel_gpu/graph/network.hpp

```diff
@@ -128,9 +128,25 @@ struct network {
             evt = get_primitive_event(output_id);
         return network_output(evt, get_output_memory(output_id), get_stream_ptr());
     }
+    layout get_node_output_layout(const primitive_id& output_id) const;
 
+    template <class T>
+    std::vector<float> get_output_values_to_float(const primitive_id& output_id, size_t max_cnt = std::numeric_limits<size_t>::max()) {
+        std::vector<float> ret;
+        auto ptr = get_output_memory(output_id);
+        auto out_ids = get_output_ids();
+        if (find(out_ids.begin(), out_ids.end(), output_id) == out_ids.end())
+            IE_THROW() << "Non output node's memory may have been reused. "
+                          "Make target node to output by using ov::intel_gpu::custom_outputs in ExecutionConfig.";
+        mem_lock<T, mem_lock_type::read> mem(ptr, get_stream());
+        if (ptr->get_layout().data_type != type_to_data_type<T>::value)
+            IE_THROW() << "target type " << data_type_traits::name(type_to_data_type<T>::value)
+                       << " mismatched with actual type " << data_type_traits::name(ptr->get_layout().data_type);
+        for (size_t i = 0; i < std::min(max_cnt, ptr->get_layout().count()); i++)
+            ret.push_back(mem[i]);
+        return ret;
+    }
```

Comment:
Oops sorry for the belated response, but this function is for test purpose. In that case, it would be better to put it in test code instead of execution code.